### PR TITLE
(DOCSP-29615): Swift: Update log level documentation for new Logger API

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

This PR depends on realm-swift PR #8199 and a subsequent release. After release:

- [x] Update [VERSION] in ln 16 and 47 
- [x] Add links to the appropriate API reference docs in ln 19 and 21
- [x] Update the test suite to the realm-swift release version instead of the feature branch

### Jira

- https://jira.mongodb.org/browse/DOCSP-29615

### Staged Changes

- [Set the Client Log Level](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-29615/sdk/swift/sync/log-level/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Review by krollins-mdb — APPROVED** [review]
LGTM, with a few non-blocking suggestions.

File: source/examples/generated/code/start/Sync.snippet.set-log-level-logger.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Access your app
+let app = App(id: YOUR_APP_SERVICES_APP_ID)
+
+var logs: String = ""
```
Comment (dianaafanador3) [inline source/examples/generated/code/start/Sync.snippet.set-log-level-logger.swift:4]: This is per testing purposes, most probably what the end user wants is to set a different format for the the log info.
For example, adding a "REALM DEBUG" string before every Realm log, so they can distinguish from their own logs, or maybe adding this log information to a file.

File: source/sdk/swift/sync/log-level.txt [hunk 2]
```diff
@@ -11,8 +11,45 @@ Set the Client Log Level - Swift SDK
...
+You can set a logger as the default logger for your app with ``Logger.shared``. 
+After you set the default logger, you can change the log level during the app 
+lifecycle as needed.
+
+.. literalinclude:: /examples/generated/code/start/Sync.snippet.set-default-logger.swift
+   :language: swift
+
+.. tip:: Default log levels
+
+   The default log level is ``.info``. To disable all log messages, set the
```
Comment (dianaafanador3) [inline source/sdk/swift/sync/log-level.txt:45]:
```suggestion
   The default Realm log level is ``.info``. To disable all log messages from Realm, set the 
```

**Review by dianaafanador3 — COMMENTED** [review]
LGTM, just a few suggestions

**Pull request merged**